tests/getenclave_extend/README.md

```diff
@@ -0,0 +1,5 @@
+This directory tests enclave creation with different type and flags, with debug and non-debug images. It has the following tests:
+* Creating an enclave with TYPE_SGX and FLAG_DEBUG;
```

Comment:
This test matrix doesn't reflect what's actually tested:
* "auto" is an enclave type specifier, not a flag (i.e. `OE_ENCLAVE_TYPE_AUTO`). It's also interesting to test in this matrix because its behavior does not interact with the type flags. The interesting dimensions to test for the type specifier would be:
    * use `OE_ENCLAVE_TYPE_AUTO` and feed it an OP-TEE enclave and an SGX enclave, it should select the appropriate type without having to parameterize the `oe_create_enclave()` call explicitly. 
    * alternatively, your test matrix would be explicitly be the choice of `OE_ENCLAVE_TYPE_SGX` and `OE_ENCLAVE_TYPE_OPTEE`
    * This is not an interesting dimension to test currently because I'm pretty sure they would not pass for OP-TEE; the test success criteria here do not apply to OP-TEE as it doesn't not have a converged concept of debug/simulate with SGX. 

* "simulate" is a flag, not an enclave type (i.e. `OE_ENCLAVE_FLAG_SIMULATE`) and it is never set in the test code together with `OE_ENCLAVE_FLAG_DEBUG` as suggested in the third case here.